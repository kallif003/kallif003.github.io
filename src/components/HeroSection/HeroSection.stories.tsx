import React from "react"
import { ComponentStory, ComponentMeta } from "@storybook/react"
import img from "../../../assets/Image/sistema-delivery.png"
import { H3Head } from "../Typography"
import { HeroSection, Container, Imagem } from "./HeroSection"

export default {
	title: "Moleculas/HeroSection",
	component: HeroSection,
} as ComponentMeta<typeof HeroSection>

const Template: ComponentStory<typeof HeroSection> = (args) => (
	<div>
		<HeroSection color={args.color}>
			<Container>
				<Imagem
					src={args.img}
					alt="Sistema Delivery"
					width={args.width}
					height={args.height}
				/>
			</Container>
			<H3Head className="text-white pl-10 pt-10">
				Sistema Delivery - Pizzarias & Esfiharias
			</H3Head>
		</HeroSection>
	</div>
)

export const Yellow = Template.bind({})
Yellow.args = {
	color: "#EFB72B",
	width: "500px",
	height: "300px",
	img: img,
}

export const Purple = Template.bind({})
Purple.args = {
	color: "#9000FF",
	width: "500px",
	height: "300px",
	img: img,
}

export const Red = Template.bind({})
Red.args = {
	color: "#C2273A",
	width: "500px",
	height: "300px",
	img: img,
}
