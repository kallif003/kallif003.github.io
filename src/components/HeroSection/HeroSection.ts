import styled from "styled-components"
import Image from "next/image"

export const HeroSection = styled.div`
	height: 534px;
	width: 1155px;
	margin-top: 16px;
	margin-bottom: 16px;
	padding-top: 7rem;
	border-radius: 20px;
	background-color: ${({ color }) => color};
	margin-left: auto;
	margin-right: auto;
`
export const Container = styled.div`
	margin-left: auto;
	margin-right: auto;
	width: 30rem;
`
export const Imagem = styled(Image).attrs((props) => ({
	width: props.width,
	height: props.height,
	quality: props.quality,
}))`
	border-radius: 10px 40px 5px;
`
