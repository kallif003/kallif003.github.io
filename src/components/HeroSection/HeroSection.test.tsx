import * as React from "react"
import { render } from "@testing-library/react"
import { HeroSection, Container } from "../HeroSection/HeroSection"

describe("Hero Section Components", () => {
	it("renders HeroSection properly", () => {
		const { container } = render(<HeroSection>Teste</HeroSection>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("renders Container properly", () => {
		const { container } = render(<Container>Teste</Container>)
		expect(container.firstChild).toMatchSnapshot()
	})
})
