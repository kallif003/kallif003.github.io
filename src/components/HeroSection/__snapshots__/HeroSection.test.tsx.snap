// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Section Components renders Container properly 1`] = `
<div
  class="sc-gsDKAQ cToFAP"
>
  Teste
</div>
`;

exports[`Hero Section Components renders HeroSection properly 1`] = `
<div
  class="sc-bdvvtL gmKzVj"
>
  Teste
</div>
`;
